src/core/viz/expressions/time.js

```diff
@@ -32,32 +32,36 @@ export default class Time extends BaseExpression {
         this.date = util.castDate(date);
         this.inlineMaker = inline => inline._impostor;
     }
-    
+
     _compile(meta) {
         this.metadata = meta;
-        const inputMin = this.metadata.columns.find(c => c.name == this.dateProperty.name).min.getTime();
-        const inputMax = this.metadata.columns.find(c => c.name == this.dateProperty.name).max.getTime();
+        const inputMin = util.castDate(this.metadata.columns.find(c => c.name).min).getTime();
```

Comment:
`c => c.name` This doesn't make sense, you are searching for a column and you are accepting the first one with a name.

Reply:
> wops, totally true, how can we get the column name from the time expression?

The parent (in this case the binary expression) should set it, check my original PR (see between)